polkadot/node/subsystem-bench/src/subsystem-bench.rs

```diff
@@ -198,6 +216,52 @@ impl BenchCli {
 	}
 }
 
+#[cfg(target_os = "linux")]
+fn is_valgrind_mode() -> bool {
```

Comment:
nit: we could add all of these functions to a linux-only `valgrind` module for better encapsulation. also, we could avoid having empty valgrind functions.

Reply:
Yes, it's good to extract it to a module, but how to avoid empty functions?

Reply:
if you add `#![cfg(target_os = "linux")` to the top of the valgrind file, it'll only be compiled on linux. Then you'd have to only call the valgrind functions on linux (add #[cfg()]s to the calling code). Then you wouldn't need empty functions